src/service.rs

```diff
@@ -81,9 +85,13 @@ pub async fn upgrade_protocol(
             }
         },
         None => {
-            let err_msg = "Missing X-Session-Id in upgrade protocol request".to_string();
-            error!(err_msg);
-            return NotaryServerError::BadProverRequest(err_msg).into_response();
+            if let Some(param_session_id) = params.session_id {
```

Comment:
Looks good! Though do you mind changing the outer logic altogether to only use query param instead of header for session id? Don't think we need to support both cases :)

Reply:
No problem. I've updated it in the latest commit